yaml-tests/src/test/resources/transactions-tests.yamsql

```diff
@@ -17,41 +17,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# you can add comments like this one to your scripts.
-#
-# The file is split into "documents" wherein each document is describes a block. A type of block exhibits a unique
-# functional behavior. At the top-level, there are 2 types of blocks:
-#   - setup_block: These blocks are used to control the environment needed to run the test blocks. For instance, they
-#       are used to "setup", "populate" and "destruct" the databases and tables. A file can have multiple of these to
-#       `tweak` the environment as needed by the testing block.
-#   - test_block: Defines a scope for a group of tests by setting the knobs that determines how those tests are run.
-#
-# The config block can be of the following types:
-#   - schema_template: This block carries the definition of the template based on which the database and schemas are setup
-#       automatically on the execution for the following `test_block` to use. Note that a file can have multiple of these
-#       blocks, each creating a database and schema with a different name scheme. Consequently, cleaning up of multiple
-#       of these blocks happens after all the blocks in the file has been executed, in the order of their definition.
-#       That is, 1st `schema_template` gets cleaned first, then second, then third and so on.
-#   - setup: This block gives more flexibility to the tester to setup the tests. Compared to `schema_template`, which are
-#       opaque as to how and where schemas are created, `setup` setup block gives the tester the option to execute
-#       whatever statements they want in whichever location.
+# This test serves as an example of using transaction setup, both via references and inline.
 ---
 options:
-    supported_version: 4.3.5.0
+    supported_version: 4.4.7.0
 ---
 schema_template:
-    create table t1(id bigint, col1 bigint, primary key(id))
+    create table table_t1(id bigint, col1 bigint, primary key(id))
+---
+# This document type creates a map of transaction setups that can be referenced in individual queries
+# This helps to reduce duplication.
+# The map is basically from a name (e.g. t1_less_than_50) to a statement to run at the start of every query.
+# They're referenced in the query using `setupReference`. Right now it only supports a single statement as a string, but
+# it could be expanded to support an array in the future.
+transaction_setups:
```

Comment:
it seems like multiple `transaction_setups` are allowed, so they can be spread all over the file. However, since all of them are processed in `parsing`, they all exist for the entire runtime. I think it will be good to disallow that or have an example citing this behavior.

Reply:
It appears to work as expected; I added some tests that cover this, and show that only `test_block`s after the `transaction_setup` can refer to it. It doesn't make a lot of sense if you have just tests, but if you have some tests, some setup, and then some transaction_setups it may make more sense.